### PR TITLE
Add js tests

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@PVince81 mind helping out to fix the failing tests - I don't understand this stuff - THX

```
$ node_modules/karma/bin/karma start karma.config.js --single-run
05 03 2018 22:38:22.821:INFO [karma]: Karma v1.7.1 server started at http://0.0.0.0:9876/
05 03 2018 22:38:22.823:INFO [launcher]: Launching browser PhantomJS with unlimited concurrency
05 03 2018 22:38:22.828:INFO [launcher]: Starting browser PhantomJS
05 03 2018 22:38:22.989:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket HSJVz6wMWOWhPvYWAAAA with id 95311285
PhantomJS 2.1.1 (Linux 0.0.0) LOG: 'JQMIGRATE: Migrate is installed, version 1.4.0'

WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
PhantomJS 2.1.1 (Linux 0.0.0) DeleteHandler tests shows a notification when marking for delete FAILED
        TypeError: undefined is not a function (evaluating 'this.$el.find('.quota-user').singleSelect()') thrown
INFO: 'sinon.reset is deprecated and will be removed from the public API in a future version of sinon.'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
WARN: 'Missing argument $row in OC.Notification.hide() call, caller needs to be adjusted to only dismiss its own notification'
PhantomJS 2.1.1 (Linux 0.0.0): Executed 9 of 9 (1 FAILED) (0.039 secs / 0 secs)

### CODE REVIEW

**Comment by PVince81** [comment]
try running with `node_modules/karma/bin/karma start karma.config.js` (no arguments) then open "localhost:9876/debug.html" in the browser and look at the console / network console.

See whether singleselect.js and/or multiselect.js libraries were loaded.

From what I see we did not mock these in core. They should normally be loaded through the "core.json" file in karma.config.js

**Comment by DeepDiver1975** [comment]
> See whether singleselect.js and/or multiselect.js libraries were loaded.

singleselect.js was missing .... no idea why this worked in core ....

File: .drone.yml [hunk 1]
```diff
@@ -93,6 +93,17 @@ pipeline:
       matrix:
         TEST_SUITE: phpunit
 
+  js-tests:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - cd /var/www/owncloud/apps/user_management/tests/js
```
Comment (PVince81) [inline .drone.yml]: makefile would be nicer

File: tests/js/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+	"name": "owncloud-activity",
```
Comment (PVince81) [inline tests/js/package.json]: not activity

File: .drone.yml (hunk 1 above)
Comment (DeepDiver1975) [reply .drone.yml]: any hint where this is done properly?

Comment (PVince81) [reply .drone.yml]: here but I don't expect that complex a Makefile for starters: https://github.com/owncloud/customgroups

let's use the way how core does it: https://github.com/owncloud/core/blob/master/Makefile#L171

also see http://www.olioapps.com/blog/the-lost-art-of-the-makefile/ if you want to make it a little bit for fancy ;-)

**Comment by DeepDiver1975** [comment]
failing phpunit tests require #5 to be merged

**Pull request merged**